File: MomentsJava/app/src/main/java/com/example/momentsjava/data/model/ListItem.java
package com.example.momentsjava.data.model;

import androidx.annotation.NonNull;

public class ListItem {
    private UserInfo userInfo;
    private MomentInfo momentInfo;

    public ListItem(UserInfo userInfo, MomentInfo momentInfo) {
        this.userInfo = userInfo;
        this.momentInfo = momentInfo;
    }

    public UserInfo getUserInfo() {
        return userInfo;
    }

    public void setUserInfo(UserInfo userInfo) {
        this.userInfo = userInfo;
    }

    public MomentInfo getMomentInfo() {
        return momentInfo;
    }

    public void setMomentInfo(MomentInfo momentInfo) {
        this.momentInfo = momentInfo;
    }

    @NonNull
    @Override
    public String toString() {
        return "ListItem{" +
                "userInfo=" + userInfo +
                ", momentInfo=" + momentInfo +
                '}';
    }
}

File: app/src/main/java/com/example/moments/data/model/ListItem.kt
package com.example.moments.data.model

data class ListItem(
    val userInfo: UserInfo,
    val momentInfo: MomentInfo
)

data class UserInfo(
    val username: String,
    val avatar: String
)

data class MomentInfo(
    val text: String,
    val picture: List<String>
)
